src/main/resources/db/migration/all/20230116143807__add_deallocated_at_to_assessments.sql

```diff
@@ -0,0 +1 @@
+ALTER TABLE assessments ADD COLUMN deallocated_at TIMESTAMP WITH TIME ZONE;
```

Comment:
did you consider using  `reallocated_at` rather than `deallocated_at`?

I'm guessing your choice is in recognition of the fact that it's not always possible to reallocate a task?

I wonder if it would clearer (more natural, more intention-revealing) to use `reallocated_at` and only set it once it **has** been possible to identify and set a suitable assessor?

Reply:
So I went with deallocated because we're reallocating the Application which results in:
- The existing Assessment being rendered disabled (deallocated?)
- A new Assessment being created which is allocated to the other user

So I think from the perspective of the original Assessment it has been deallocated, or perhaps `disabled` or `superseded`?

Reply:
Just noticed I missed this point:

> I'm guessing your choice is in recognition of the fact that it's not always possible to reallocate a task?

As it stands, if it's not possible to allocate to the provided user the current Assessment remains active.  The whole thing is also transactional so any database exceptions should result in keeping the existing state.

Reply:
OK, that behaviour makes sense. 👍  So it sounds like it's purely a (more or less "bike-shedding") question of naming: de-allocate versus re-allocate.

But, do you think we should have a test to make it clear that when reallocation fails the assessment does not go into the deallocated state?

Reply:
Have renamed to reallocated and added an integration test to ensure the operation is amotic